src/server/api/controllers/messages-controller.js

```diff
@@ -1,10 +1,46 @@
+/* eslint-disable @typescript-eslint/camelcase */
+/* eslint-disable camelcase */
 const knex = require('../../config/db');
 const Error = require('../lib/utils/http-error');
 const moment = require('moment-timezone');
 
-const getMessages = async () => {
+const getMessages = async (query, channel_id, sender, limit, sort, sort_by) => {
+  const getMes = knex('channel_messages')
+    .select(
+      knex.raw(
+        'DISTINCT channel_messages.id, channel_messages.message, channel_messages.fk_channel_id,channel_messages.fk_user_id,channel_messages.created_at, channel_messages.updated_at,channel_messages.deleted_at,users.id,users.profile_image, users.user_name,users.email, users.last_seen, users.created_at, users.updated_at, users.deleted_at',
+      ),
+    )
+    .leftJoin(knex.raw('users ON users.id = channel_messages.fk_user_id'));
```

Comment:
You dont need to do a join of the users. You can just do a where select on the fk_user_id. Ahh its because i wrote that you could select messages from the email. Just drop that. Only make the possibility for getting a users messages using the user id.